lib/Optimizer/Transforms/GenKernelExecution.cpp

```diff
@@ -282,6 +282,9 @@ class GenerateKernelExecution
     auto thunkTy = cudaq::opt::marshal::getThunkType(ctx);
     auto thunk =
         builder.create<func::FuncOp>(loc, classNameStr + ".thunk", thunkTy);
+
+    thunk->setAttr("noinline", builder.getUnitAttr());
+    thunk->setAttr("optnone", builder.getUnitAttr());
```

Comment:
I'm not sure I follow why we're changing when inlining is done *and* changing what gets inlined.

IIRC, the issue was that a kernel was being inlined into its thunk function before it was synthesized, which meant the synthesis was happening too late. If we're changing the order there, do synthesis and then do inlining, it no longer matters if the kernel is inlined into the thunk, right?

Reply:
Also, disabling *any* optimization on a thunk function seems punitive. What is the justification for doing that?

Reply:
I think I can remove those as they don't seem to work anyway, and the problem is solved by reordering the passes.

Details:  It was me trying to not inline the entry point function into the trunk before the `ReturnToLogOutput` pass. AFAIK the inlining is done by the canonicalize (surprisingly, not the inline pass) so I had to try `no opt` as well. Neither of those attributes seem to help, so I moved the aggressive inlining pass out of `commonPipelineConvertToQIR` (we already have it in all the pipelines for quantum devices, and I had to add it separately in `cudaq-translate` (for QIR profiles).

Reply:
Removed, thanks!